pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java

```diff
@@ -414,5 +418,9 @@ public Authentication getProxyClientAuthenticationPlugin() {
         return this.proxyClientAuthentication;
     }
 
+    public void addPrometheusRawMetricsProvider(PrometheusRawMetricsProvider metricsProvider) {
+        metricsProviders.add(metricsProvider);
+    }
```

Comment:
Since this method could be called from other threads in protocol handlers, I think we should consider using a concurrent data structure to back the `metricsProviders` list. Technically, it appears that the current implementation in the broker does not use a concurrent data structure, so I could be wrong about this class's thread safety requirements.

Reply:
@codelipenghui as you contributed the code for the broker, can you say if we should use concurrent structures for `PulsarService`'s `pendingMetricsProviders` ?

Reply:
Initially, it was only considered when loading plugins when starting the broker, which will not be modified after the broker started. To avoid any potential thread-safety issues(maybe some users want to modify after the broker started), using concurrent structures looks good to me.